crates/rpc-types-eth/src/filter.rs

```diff
@@ -538,6 +538,74 @@ impl Filter {
     pub fn has_topics(&self) -> bool {
         self.topics.iter().any(|t| !t.is_empty())
     }
+
+    /// Checks if a given Log matches this Filter.
+    ///
+    /// This function checks:
+    /// 1. If the log's block number/hash matches the filter's block option (if any)
+    /// 2. If the log's address matches the filter's address (if any)
+    /// 3. If the log's topics match the filter's topics (if any)
+    ///
+    /// Returns true if the Log matches the Filter, false otherwise.
+    pub fn matches_log(&self, log: &RpcLog<LogData>) -> bool {
+        // Check if block option matches
```

Comment:
let's destructure `self` here like `let Self { a, b, c } = self` to make sure we're using all of the fields.